spec/std/yaml/from_yaml_spec.cr

```diff
@@ -0,0 +1,374 @@
+require "spec"
+require "yaml"
+
+private struct ObjectStub
+  getter called : Bool
+
+  def self.new(pull : YAML::PullParser)
+    pull.read_scalar
+    new(true)
+  end
+
+  def initialize(@called)
+  end
+end
+
+describe "Object.from_yaml" do
+  it "should create call .new on with a YAML::PullParser" do
+    ObjectStub.from_yaml("hello").called.should be_true
+  end
+end
+
+describe "Nil.from_yaml" do
+  it "should serialize and deserialize" do
+    Nil.from_yaml(nil.to_yaml).should eq nil
+  end
+
+  values = {"", "NULL", "Null", "null", "~"}
+
+  it "should return nil if a YAML null value" do
+    values.each do |value|
+      # Test with an array since a standalone empty value shows as STREAM_END
+      Array(Nil).from_yaml("- " + value).should eq [nil]
+    end
+  end
+
+  it "should raise if a YAML null value is quoted with double quotes" do
```

Comment:
To further simplify specs a little bit, I'd recommend to merge single and double quoted examples together and call it `should raise if a YAML null value is quoted` (etc.). There is no real point in having separate specs. It's about to test the same behaviour, quoted values, which just comes in two different variations.